shell/common/engine.cc

```diff
@@ -478,24 +478,30 @@ void Engine::ScheduleSecondaryVsyncCallback(const fml::closure& callback) {
 }
 
 void Engine::HandleAssetPlatformMessage(fml::RefPtr<PlatformMessage> message) {
-  fml::RefPtr<PlatformMessageResponse> response = message->response();
-  if (!response) {
-    return;
-  }
-  const auto& data = message->data();
-  std::string asset_name(reinterpret_cast<const char*>(data.data()),
-                         data.size());
-
-  if (asset_manager_) {
-    std::unique_ptr<fml::Mapping> asset_mapping =
-        asset_manager_->GetAsMapping(asset_name);
-    if (asset_mapping) {
-      response->Complete(std::move(asset_mapping));
-      return;
-    }
-  }
-
-  response->CompleteEmpty();
+  fml::TaskRunner::RunNowOrPostTask(
```

Comment:
There is no need for `RunNowOrPostTask`, the IO and UI task runners are never identical and you don't need this to be synchonous.